hexa/pipelines/models.py

```diff
@@ -288,6 +290,13 @@ def update_if_has_perm(self, principal: User, **kwargs):
             if key in kwargs:
                 setattr(self, key, kwargs[key])
 
+        if (
```

Comment:
That's not easy to read and is not optimal. In the case where you have disabled the webhook and enabled it again it wouldn't recreate a token.

What do you think of a `set_webhook_state(enabled: bool)` on the `Pipeline` ? You could call this method if the `webhook_enabled` is in the kwargs